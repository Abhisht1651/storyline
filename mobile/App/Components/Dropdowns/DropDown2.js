// @flow

import React, { useEffect } from 'react'
import { View, Text } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import styles from '../Styles/DropDownStyles/DropDown2Style'
import IconFilter from '../../Images/Icons/Icon-filter.svg'
import IconCheckmark from '../../Images/Icons/icon-checkmark-purple.svg'
import IconDropDown from '../../Images/Icons/Icon-dropdown.svg'
import useHooks from './DropDown2Hooks'

type Props = {
  headerTitle: string,
  type: number,
  content: Array<string>,
  getOption: (arg: string) => void,
  zIndex: number
}

const DropDown2 = (props: Props) => {
  const { headerTitle, type, content, getOption, zIndex } = props
  const { hidden, selected, toggleHandler, clickHandler } = useHooks()

  useEffect(() => {
    getOption(selected)
  }, [selected])

  const shadowStyle = {
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowColor: '#1F1F6016',
    shadowOffSet: { width: 0, height: 0 }
  }

  const list = []
  for (let i = 0; i < content.length; i += 1) {
    if (i !== content.length - 1) {
      list.push(
        <TouchableOpacity
          onPress={() => clickHandler(i)}
          style={styles.dropDownWidth}
          activeOpacity={1}>
          <View style={styles.listItem}>
            <Text style={styles.itemText}>{content[i]}</Text>
            <TouchableOpacity style={styles.checkBox}>
              {selected === i + 1 && (
                <View style={styles.checkedBox}>
                  <IconCheckmark />
                </View>
              )}
            </TouchableOpacity>
          </View>
        </TouchableOpacity>
      )
    } else {
      list.push(
        <TouchableOpacity
          onPress={() => clickHandler(i)}
          style={styles.dropDownWidth}
          activeOpacity={1}>
          <View style={styles.listLastItem}>
            <Text style={styles.lastItemText}>{content[i]}</Text>
            <TouchableOpacity style={styles.checkBox}>
              {selected === i + 1 && (
                <View style={styles.checkedBox}>
                  <IconCheckmark />
                </View>
              )}
            </TouchableOpacity>
          </View>
        </TouchableOpacity>
      )
    }
  }

  return (
    <View style={styles.container} zIndex={zIndex}>
      <TouchableOpacity onPress={toggleHandler} style={styles.iconOnly} activeOpacity={1}>
        {type === 1 ? <IconFilter /> : <IconDropDown />}
      </TouchableOpacity>

      {!hidden && (
        <View style={[styles.dropDownBody, shadowStyle]} elevation={4}>
          <View style={styles.dropDownHeader}>
            <TouchableOpacity style={styles.headerItem}>
              <Text style={styles.headerText}>{headerTitle}</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={toggleHandler} style={styles.dropDownIcon} activeOpacity={1}>
              {type === 1 ? <IconFilter /> : <IconDropDown />}
            </TouchableOpacity>
          </View>
          {list}
        </View>
      )}
    </View>
  )
}

DropDown2.defaultProps = {
  headerTitle: 'Filter by',
  type: 1,
  content: ['All', 'Sent', 'Received'],
  zIndex: 1
}

export default DropDown2
