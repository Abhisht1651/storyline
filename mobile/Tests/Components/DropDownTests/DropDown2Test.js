import { shallow } from 'enzyme'
import React from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

import useHooks from '../../../App/Components/Dropdowns/DropDown2Hooks'
import DropDown2 from '../../../App/Components/Dropdowns/DropDown2'

describe('DropDown2', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<DropDown2 />)
      expect(component).toMatchSnapshot()
    })
  })
})

describe('DropDown2', () => {
  it('should hide/show the dropdown', () => {
    const { result } = renderHook(useHooks)
    const val = result.current.hidden
    act(() => {
      result.current.toggleHandler()
    })
    expect(result.current.hidden).toBe(!val)
  })

  it('should select/unselect the option', () => {
    const { result } = renderHook(useHooks)
    act(() => {
      result.current.clickHandler(1)
    })
    expect(result.current.selected).toBe(2)

  })
})
