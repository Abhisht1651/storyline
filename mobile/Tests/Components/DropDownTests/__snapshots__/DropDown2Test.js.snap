// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown2 Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
    }
  }
  zIndex={1}
>
  <TouchableOpacity
    activeOpacity={1}
    delayLongPress={600}
    extraButtonProps={
      Object {
        "rippleColor": "transparent",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <SvgMock />
  </TouchableOpacity>
</View>
`;
